lib/core/mergeConfig.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var utils = require('../utils');
+
+/**
+ * Config-specific merge-function which creates a new config-object
+ * based on given defaults and instance config.
+ *
+ * @param {Object} config1
+ * @param {Object} config2
+ * @returns {Object} New object resulting from merging config2 to config1
+ */
+module.exports = function mergeConfig(config1, config2) {
+  // eslint-disable-next-line no-param-reassign
+  config2 = config2 || {};
+  var config = {};
+
+  utils.forEach(['url', 'method', 'params', 'data'], function valueFromInstanceConfig(prop) {
+    if (typeof config2[prop] !== 'undefined') {
+      config[prop] = config2[prop];
+    }
+  });
+
+  utils.forEach(['headers', 'auth', 'proxy'], function mergeInstanceConfigWithDefaults(prop) {
+    if (typeof config2[prop] !== 'undefined') {
```

Comment:
Can we simplify it like this?

```js
function mergeValues(prop) {
  if (config1[prop] !== 'undefined' || config2[prop] !== 'undefined') {
    config[prop] = utils.deepMerge(config1[prop], config2[prop]);
  }
}
```

It should be ok to pass `null` or `undefined` to `deepMerge` so I would just check if at least one of the config objects contains the prop.

Reply:
I would also change the name to smth like `mergeValues`